usr/share/rear/lib/authtoken-functions.sh

```diff
@@ -0,0 +1,546 @@
+#!/bin/bash
+
+function authtkn_load() {
+	# $1 container path
+	# $2 container offset
+	# $3 encryption key
+	# $4 [bound_to_blockdev path]
+
+	local intermediate_key=""
+	local cipher
+	local plain
+	local ret
+
+	cipher=$(token_read "$1" "$2"); ret=$?
+	if [ $ret -ne 0 ]; then
+		echo "authtkn_load(): token_read() failed: $ret" >&2
+		return $ret
+	fi
+
+	if [ -n "$4" ]; then
+		intermediate_key=$(blkdev_hash "$4" 2); ret=$?
+		if [ $ret -ne 0 ]; then
+			echo "authtkn_load(): failed to derive intermediate key from $4: $ret" >&2
```

Comment:
Don't use plain 'echo' for messages.
Use an appropriate ReaR function for that, cf.
https://github.com/rear/rear/wiki/Coding-Style#what-to-do-with-stdin-stdout-and-stderr

Reply:
Ah - yes - again I missed that this is run during
recovery system startup - so 'echo' is right here.

But 'echo' is not right when those functions are also run
by sbin/rear (e.g. "rear mkrescue/mkbackup" and "rear recover").

As far as I see the problem is that we cannot source
lib/_input-output-functions.sh during recovery system startup, see
https://github.com/rear/rear/issues/2955#issuecomment-1465642166
```
... we cannot 'source lib/_input-output-functions.sh'
during ReaR recovery system startup in
skel/default/etc/scripts/system-setup
because lib/_input-output-functions.sh also does special stuff
like stdin stdout redirections and exit task settings.
I think this should not be in a script that defines functions
but it was always therein and I do not understand
why this is in lib/_input-output-functions.sh and
not in sbin/rear (where I would expect it)
so I left it as is because it works as is.
```

Reply:
Only a totally offhanded idea:

Perhaps we may provide for the recovery system startup scripts
simplified input-output functions with same names as we already
have in lib/_input-output-functions.sh
```
Print
UserOutput
PrintError
Log
Debug
DebugPrint
LogPrint
LogUserOutput
LogPrintError
UserInput
```
so "dual environment" code (i.e. code that is meant to run
both during recovery system startup and also in sbin/rear)
could call them same?

Reply:
Yes, I'd love to see that. I think it is actually feasible to turn `/etc/scripts/system-setup` into a mini-ReaR that would share the same configuration tree but run different scripts. Maybe we just convert that into a `system-setup` workflow and that's it. Replace calling that script with `rear system-setup` and we are done.

Reply:
Although `authtoken-functions.sh` are generic enough to possibly find some use also in ReaR context over time, I'd prefer to keep them self-contained and easily reusable outside of ReaR